File: dev-docs/snippets/auth/get-app-config.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fetch the app-launch config subset (mode, waitlist, available auth methods).
export async function getAppConfig(client: JsBaoClient) {
  // #region example
  const config = await client.getAppConfig();
  if (config.magicLinkEnabled) {
    // offer magic-link sign-in
  }
  // #endregion example
  return config;
}

File: dev-docs/snippets/auth/get-app-config.swift
import JsBaoClient

// Fetch the app-launch config subset via the typed `client.auth` namespace.
// Returns a typed `AppConfigInfo` (`appId`, `name`, `mode`, `waitlistEnabled`,
// `hasOAuth`, `hasPasskey`, `magicLinkEnabled`).
func getAppConfig(client: JsBaoClient) async throws {
  // #region example
  let config = try await client.auth.getAppConfig()
  let magicLinkEnabled = config.magicLinkEnabled
  // #endregion example
  _ = magicLinkEnabled
}
